molgenis-jobs/src/main/java/org/molgenis/data/jobs/Job.java

```diff
@@ -31,18 +31,22 @@ public Job(Progress progress, TransactionTemplate transactionTemplate, Authentic
 	@Override
 	public Result call()
 	{
+		if (transactionTemplate != null)
+		{
+			return transactionTemplate.execute((status) -> doCallInTransaction());
+		}
+		else
+		{
+			return doCallInTransaction();
```

Comment:
The Jobs run running on a separate executor thread.
If we do nothing, it will run without permissions and not in a transaction.
It is no bean so annotating the call() method with @Transactional won't do anything.
So we need to manually set the permissions and manually call the TransactionTemplate.

This PR turns the order inside out.
Call the transaction template within the user switch.
Used to be switch the user within the transaction.
The bug was that the afterCommit then runs after the permissions get restored.